src/psyclone/tests/dynamo0p3_test.py

```diff
@@ -1996,6 +1998,10 @@ def test_r_solver(tmpdir):
     appropriate precision in the PSy-layer and all required constants
     are declared.
 
+    This test could be removed once issues #1638/#1667 are addressed
```

Comment:
How about "TODO" here? Regarding the issue #1638 mentioned here, the support for `r_solver_field` is on LFRic trunk. I am happy to provide the files since you have issues with access to MOSRS.

Reply:
Changed to a TODO.

Reply:
If you could provide me with the r_solver_field infrastructure file(s) for #1638, that would be great.

Reply:
> If you could provide me with the r_solver_field infrastructure file(s) for #1638, that would be great.

@rupertford, I've decided to wait a bit as all operator types are now in the same module on trunk, and Mike has a ticket in review with Chris to separate them (https://code.metoffice.gov.uk/trac/lfric/ticket/3144).